doc/smart-switch/PA-Validation/SmartSwitchPAValidationOffload.md

```diff
@@ -0,0 +1,149 @@
+  - [2.3 DPU Shut/Restart](#23-dpu-shutrestart)
+- [3. Test Plan](#3-test-plan)
+
+###### Revision
+
+| Rev |     Date    |       Author          | Change Description                |
+|:---:|:-----------:|:---------------------:|-----------------------------------|
+| 0.1 |  06/10/2024 |     Kumaresh Perumal  | Initial version                  |
+| 0.2 |  08/21/2024 |          Yakiv Huryk  | Added DashOffloadManager         |
+
+
+# About this Manual
+This document provides general information about offloading PA validation feature to NPU in Smartswitch.
+
+# Definitions/Abbrevations
+
+|                          |                                          |
+|--------------------------|------------------------------------------|
+| ACL                      | Access Control List                      |
+| NPU                      | Network Processing Unit                  |
+| DPU                      | Data Processing Unit                     |
+ 
+
+# 1. Requirements Overview
+
+## 1.1 Functional Requirements
+- Provide a generic infrastructure for offloading DASH functionality to the NPU (DashOffloadManager).
+- DashOffloadManager should listen to the DASH configuration and either intercept it for offloading, or forward it to the DPU.
+- Provide PA validation support in NPU when this feature is not supported in DPU pipeline model.
+- Use existing ACL tables and ACL rules design in NPU.
+
+## 1.2 Scale Requirements
+
+# 2 Modules Design
+
+## 2.1 Dash Offload Manager
+The new orchagent application DashOffloadManager will be responsible for DASH offloading logic. It will collect all the needed information for offloading and perform all the relevant configurations.
+To get the DASH configuration that should be offloaded, the DashOffloadManager will act as a transparent ZMQ proxy between the GNMI server and the DPU swss, forwarding all the configuration and intercepting the tables that should be offloaded.
+To simplify the management of the DPU offload and achieve optimal performance, every DPU is handled by a separate instance of a ZMQ Proxy (pair of ZMQ Server and Client)
+
+<img src="images/DashOffloadManager.svg">
+
+Once the offload is required, the DashOffloadManager will start designated orch (e.g. PAValidationOffloadOrch) that will subscribe to the configuration and do the offload logic.
+
+<img src="images/DashOffloadManagerWithConsumer.svg">
+
+The Dash Offload Manager is disabled by default and only enabled for specific platforms that require its functionality.
+
+### 2.1.1 SONiC GNMI server changes
+A new parameter `zmq_dpu_proxy_address_base` is added to the telemetry.go to enable the GNMI Server -> DASH Offload Manager ZMQ connection. If specified, the GNMI server will use `zmq_dpu_proxy_address_base + dpu_index:zmq_port` as a destination for DPU ZMQ connection. For example, on the system with 2 DPUs and base address 127.0.0.10, the DASH Offload Manager's ZMQ server runs on 127.0.0.10:8100 for DPU0 and on 127.0.0.11:8100 for DPU1.

```

Comment:
Currently GNMI already get DPU address by ConfigDB "DPUS" table.
Why make this change to get DPU by base address?

https://github.com/sonic-net/sonic-gnmi/blob/e8449250bec233c81cf4b6aa324c26da7fea5732/sonic_data_client/mixed_db_client.go#L110